pac4j-cas/src/main/java/org/pac4j/cas/client/rest/CasRestAuthenticator.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2012 - 2015 pac4j organization
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.pac4j.cas.client.rest;
+
+import org.apache.commons.httpclient.HttpStatus;
+import org.jasig.cas.client.validation.TicketValidator;
+import org.pac4j.http.credentials.UsernamePasswordCredentials;
+import org.pac4j.http.credentials.authenticator.Authenticator;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+/**
+ * This is {@link CasRestAuthenticator}.
+ *
+ * @author Misagh Moayyed
+ * @since 1.8.0
+ */
+public class CasRestAuthenticator implements Authenticator<UsernamePasswordCredentials> {
+    private final URL endpointURL;
+    private final TicketValidator ticketValidator;
+
+    public CasRestAuthenticator(final URL endpointURL, final TicketValidator ticketValidator) {
+        this.endpointURL = endpointURL;
+        this.ticketValidator = ticketValidator;
+    }
+
+    @Override
+    public void validate(final UsernamePasswordCredentials credentials) {
+        final String ticketGrantingTicketId = requestTicketGrantingTicket(credentials.getUsername(), credentials.getPassword());
+        final HttpTGTProfile profile = new HttpTGTProfile(ticketGrantingTicketId);
```

Comment:
The identifier must be unique for the identity provider, it's the identifier (example: 1234)
To get a unique identifier accross all identity providers, it's the typed identifier (example: FacebookProfile#1234).

It's used to rebuild a identity and rebuild a profile if necessary. It would be best to have one, the username seems to have here the right properties (unique).